docs/intro/validator.rst

```diff
@@ -0,0 +1,225 @@
+=========
+Validator
+=========
+
+Getting started
+===============
+Validator is an instrument to unify the process of model training and validation.
+There are two main reasons to wrap your model using classes from `Validator` submodule:
+
+- make the structure of the model lifecycle clear,
+- provide API to automized model training and validation.
+
+All you need is:
+
+- define class inherited from `Validator`,
+- configure it by `yaml` file,
+- execute the following code: ::
+
+    from model_api import MyValidator
+
+    val = MyValidator('validator.yaml')
+    val.run()
+    print(val.metrics)
+
+Basic example
+=============
+
+model_api.py
+------------
+::
+
+    import ...
+
+    class PascalValidator(Validator):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.model_config = ...
+            self.ds = PascalSegmentation(bar=True)
+
+        def train_loader(self, **kwargs):
+            return self.ds.train
+
+        def test_loader(self, **kwargs):
+            return self.ds.test
+
+        def train(self, ds):
+            train_ppl = ... << ds
+            train_ppl.run()
+            self.model = train_ppl.get_model_by_name('model')
+
+        def inference(self, ds):
+            model = self.model
+            test_ppl = ... << ds
+            test_ppl.run()
+            targets = np.array(test_ppl.v('targets'))
+            predictions = np.array(test_ppl.v('predictions'))
+            return targets, predictions
+
+
+validator.yaml
+--------------
+::
+
+    train:
```

Comment:
Why is train empty?

Reply:
`train` has one input `dataset` which is created in `load_train_dataset`. We don't need to define any parameters so we define empty `train` stage in config.